unicodetools/org/unicode/tools/UpdateJspFiles.java

```diff
@@ -0,0 +1,142 @@
+package org.unicode.tools;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+import org.unicode.cldr.draft.FileUtilities;
+import org.unicode.props.IndexUnicodeProperties;
+import org.unicode.propstest.CopyPropsToUnicodeJsp;
+import org.unicode.propstest.ListProps;
+import org.unicode.text.tools.GenerateSubtagNames;
+import org.unicode.text.utility.Settings;
+
+import com.ibm.icu.util.VersionInfo;
+
+public class UpdateJspFiles {
+    static CopyOption[] options = new CopyOption[] {StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.COPY_ATTRIBUTES};
+    static final Path JSP_RESOURCE_DATA = Paths.get(Settings.UnicodeTools.UNICODEJSPS_DIR,
+    		"src/main/resources/org/unicode/jsp/");
+    static final Path UNICODE_TOOLS_DIR = Paths.get(Settings.UnicodeTools.UNICODETOOLS_DIR);
+    static final Path TRIM_PARENT = Paths.get(Settings.UnicodeTools.UNICODETOOLS_REPO_DIR);
+
+    static final String trim(Path p) {
+    	if(p.startsWith(TRIM_PARENT)) {
+    		return "{...}/"+p.subpath(TRIM_PARENT.getNameCount(), p.getNameCount()).toString();
+    	} else {
+    		return p.toString();
+    	}
+    }
+
+	public static void main(String args[]) throws IOException {
+        generateSubtagNames(); // this takes a sec, so run it first
+
+        IndexUnicodeProperties latest = IndexUnicodeProperties.make();
+        VersionInfo ucdVersion = latest.getUcdVersion();
+        System.out.println("Updating all JSP files for " + ucdVersion + " into " + TRIM_PARENT.toAbsolutePath());
+
+        copyTextFiles(ucdVersion);
+
+        copyOtherProps(ucdVersion);
+
+        // Sublaunch listProps
+        System.out.println("Sublaunching ListProps..");
+        ListProps.main(args);
+
+        // Sublaunch listProps
```

Comment:
```suggestion
        // Sublaunch CopyPropsToUnicodeJsp
```
(these comments don't really add value...)